packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -0,0 +1,136 @@
+import type { RestrictedControllerMessenger } from '@metamask/base-controller';
+import { BaseControllerV2 } from '@metamask/base-controller';
+import type { Patch } from 'immer';
+
+const controllerName = 'QueuedRequestController';
+
+// should serialize and persist these values.
+const stateMetadata = {
+  queue: { persist: false, anonymous: false },
+};
+
+const getDefaultState = () => ({});
+
+export const QueuedRequestControllerActionTypes = {
+  enqueueRequest: `${controllerName}:enqueueRequest` as const,
+};
+
+export const QueuedRequestControllerEventTypes = {
+  stateChange: `${controllerName}:stateChange` as const,
+  countChanged: `${controllerName}:countChanged` as const,
+};
+
+export type QueuedRequestControllerState = {
+  [k: string]: any;
```

Comment:
Is there a better type we can choose here rather than `any` so we don't infect this file and anything else that wants to use this controller?

Reply:
was this resolved? Seems like the type is still `any`?

Reply:
ah woops yeah missed a spot

Reply:
okay fixed now.